parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -408,21 +408,13 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener callback) {
     public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
         log.debug("Partitions revoked {}, state: {}", partitions, state);
         isRebalanceInProgress.set(true);
-        while (this.producerManager.map(ProducerManager::isTransactionCommittingInProgress).orElse(false))
+        while (isTransactionCommittingInProgress())
             Thread.sleep(100); //wait for the transaction to finish committing
 
         numberOfAssignedPartitions = numberOfAssignedPartitions - partitions.size();
 
         try {
             // commit any offsets from revoked partitions BEFORE truncation
-            this.producerManager.ifPresent(pm -> {
-                try {
-                    pm.preAcquireOffsetsToCommit();
-                } catch (Exception exc) {
-                    throw new InternalRuntimeException(exc);
-                }
-            });
-
```

Comment:
I'm guessing this was the culprit?

Reply:
yes, kind of - on its own - its not a big deal to have it here - but because it is done without check for transactional producer option - this was called even when non-transactional producer was specified - but the `pm.preAcquireOffsetsToCommit();` is only specifically for Transactional producer handling... so it was causing issue by setting write lock which was not released as committer was not transactional producer one...

Reply:
but it is not really required as `commitOffsetsThatAreReady();` few lines of code down already calls `preAcquireOffsetsToCommit()` as part of commit process.